core/web/eth_keys_controller.go

```diff
@@ -117,30 +153,40 @@ func (ekc *ETHKeysController) Create(c *gin.Context) {
 		jsonAPIError(c, http.StatusInternalServerError, err)
 		return
 	}
-	r := presenters.NewETHKeyResource(key, state,
-		ekc.setEthBalance(c.Request.Context(), state),
-		ekc.setLinkBalance(c.Request.Context(), state),
-		ekc.setKeyMaxGasPriceWei(state, key.Address),
-	)
+
+	c.Set("key", key)
+	c.Set("state", state)
 
 	ekc.app.GetAuditLogger().Audit(audit.KeyCreated, map[string]interface{}{
 		"type": "ethereum",
 		"id":   key.ID(),
 	})
-
-	jsonAPIResponseWithStatus(c, r, "account", http.StatusCreated)
 }
 
 // Delete an ETH key bundle (irreversible!)
 // Example:
 // "DELETE <application>/keys/eth/:keyID"
 func (ekc *ETHKeysController) Delete(c *gin.Context) {
 	ethKeyStore := ekc.app.GetKeyStore().Eth()
-	var err error
 
-	keyID := c.Param("keyID")
+	keyID := c.Param("address")
 	if !common.IsHexAddress(keyID) {
-		jsonAPIError(c, http.StatusBadRequest, errors.Errorf("invalid keyID: %s, must be hex address", keyID))
+		jsonAPIErrorBadRequest(c, errors.Errorf("invalid keyID: %s, must be hex address", keyID))
+		return
+	}
+
+	key, err := ethKeyStore.Get(keyID)
+	if err != nil {
+		switch err {
+		case keystore.ErrKeyNotFound:
```

Comment:
Let's use `errors.Is` to stay compatible with any wrapping.
```suggestion
		if errors.Is(err, keystore.ErrKeyNotFound)
```
And what if it doesn't match? Don't we still need to abort for other errors?